src/libstd/ffi/os_str.rs

```diff
@@ -277,6 +277,18 @@ impl OsStr {
         self.to_bytes().and_then(|b| CString::new(b).ok())
     }
 
+    /// Returns true if the `other` is a prefix of the `OsStr`.
+    #[unstable(feature = "os_str_compare", reason = "recently added")]
+    pub fn starts_with<S: AsRef<OsStr>>(&self, other: S) -> bool {
+        self.bytes().starts_with(other.as_ref().bytes())
```

Comment:
@SimonSapin? I think these should be fine. If I'm reading the "spec" correctly, WTF-8 is a strict subset of "generalized UTF-8" (https://simonsapin.github.io/wtf-8/#generalized-utf_8) and, from what I can tell, generalized UTF-8 is a prefix-free code.